src/hooks/useTypedSignTrustCredential.test.ts

```diff
@@ -0,0 +1,128 @@
+import { act, waitFor } from '@testing-library/react';
+import { useAccount, usePublicClient, useSignTypedData } from 'wagmi';
+
+import { useTypedSignTrustCredential } from './useTypedSignTrustCredential';
+import { renderHook } from '../utils/test-utils';
+import { VALID_ACCOUNT_1, VALID_ACCOUNT_2 } from '../utils/test-utils/input';
+
+jest.mock('wagmi', () => ({
+  useAccount: jest.fn(),
+  usePublicClient: jest.fn(),
+  useSignTypedData: jest.fn(),
+  useChainId: () => 1,
+}));
+
+describe('useTypedSignTrustCredetial', () => {
+  let mockUseAccount: jest.Mock;
+  let mockUsePublicClient: jest.Mock;
+  let mockUseSignTypedData: jest.Mock;
+
+  const normalSignFlow = async () => {
+    const mockSignTypedData = jest.fn();
+    mockUseSignTypedData.mockReturnValue({
+      data: 'signature',
+      signTypedData: mockSignTypedData,
+    });
+    mockSignTypedData.mockResolvedValue(true);
+
+    const { result } = await act(() =>
+      renderHook(() => useTypedSignTrustCredential()),
+    );
+
+    const { submitTypedSignRequest } = result.current;
+
+    await waitFor(
+      async () =>
+        await waitFor(() => submitTypedSignRequest(VALID_ACCOUNT_2, true)),
+    );
```

Comment:
To make `isVerified` state updated, we need to wrap waitFor twice, because the first time `waitFor` will completed with signature return, once signature return, useEffect() will trigger and setIsVerified(),  that is why we need to wrap twice to make the state completely updated.